### ISSUE TITLE
Potential to Bring Electrostatic Calculations to MolecularNodes

### ISSUE THREAD

**Issue by BradyAJohnston** [PabloEnmanuelRamos/BioBlender21#16]
I haven't looked deep into the codebase yet, but from what I can gather you use `pdb2pqr` to calculate the electrostatics of a system that you can then use for display & creation of field lines etc. 

It's not possible l don't think to perform these kinds of calculations inside of Geometry Nodes, but in theory if `pdb2pqr` just takes a `.pdb` file or something similar, then it would be possible to utilise it inside of [Molecular Nodes](https://github.com/BradyAJohnston/MolecularNodes).

If you would want to potentially work on such a project, or give me some advice & direction on how to best utilise `pdb2pqr`, then this could potentially be integrated and utilised inside of Molecular Nodes.

**Comment by MonZop** [comment]
Hi Brady, in BioBlender it was actually a bit more complex: pdb2pqr was used to place charges in the pdb file, which was then sent to APBS to calculate the potentials in the space around the molecule, and return a grid of values. Next, another custom-built program selected the (charged) atoms on the protein surface, calculated the lines (starting from the atm up to the point of potential = 0), and drew the lines with direction positive to negative. In Blender the lines were used to drive particles flowing along them in 20 or 25 frames. For moving proteins (either changing conformation or changing position) the process was repeated every 5 frames, so that the particles produced a small 'river' of charge.

**Comment by BradyAJohnston** [comment]
Hmm, it certainly does seem like it might be a bit more complex than would be possible to implement. Potentially the import of the calculated output of APBS could be done, so that the charge import and drawing of the field lines could be handled by Molecular Nodes. I think it's a powerful visualisation so enabling it would be useful.

**Comment by MonZop** [comment]
I agree that it is a powerful representation, however it is also really cumbersome to calculate and display.
I have been thinking of a possible solution, that could result in a 'fake' representation, but close enough to give some useful indication.
The idea is: build a sphere around all charges on the protein surface (or all charged aa, it depends on how easy/difficult is the selection).  Lines are drawn between the atom and the sphere border, and particles or something similar flow along the lines, positive to negative. Intensity of the charge can be seen by a) the radius of the sphere, and/or b) the number of lines.
A collision detector, when two spheres get close, triggers a different behavior of the lines/particles: if opposite charges, they all go towards the point of contact; if same charge, they deviate, and follow the surface of the sphere of origin.
When the two charged atoms get really close (establish a ionic bond), then the system goes off: no particles, no lines and no sphere.

This is all in my imagination: I am not sure that it would work in a decent way. To know, an experiment should be done using the system just described and in parallel a system that calculates proper charges and their deviation, using a true APBS or Delphi program. Things like Calcium binding, Phosphorylation or DNA binding would make great test systems.

Needless to say, all this is well beyond my personal possibilities, as I can't code and don't know enough about chemistry, nor physics :))
